Changes for Displaying Job logs in Job Details

ganeshcse2991 opened the pull request:
Display Job logs on the job details page. It will help to easily view the logs associated with the job while processing it.

_NOTE: Bull Queue version updated to the latest 3.12.1 since the older version does not support adding or getting logs for the Job._

![image](https://user-images.githubusercontent.com/7354687/72053305-c2cb5e80-32ec-11ea-8bae-006c8d50a463.png)

ttacon commented on src/server/views/dashboard/queueJobsByState.js:
```diff
@@ -96,6 +96,11 @@ async function _html(req, res) {
     jobs = jobs.filter((job) => job);
   } else {
     jobs = await queue[`get${_.capitalize(state)}`](startId, endId);
+    jobs.map(async (job) => {
```
You'll need to await on the top level returned promises from the `map` call here, otherwise it's not guaranteed that the logs will be retrieved for each job.

ttacon commented:
Thanks for the PR! Only one comment and then this is probably good to go!

ganeshcse2991 commented:
@ttacon  Addressed the review comment and made changes for it.

Changes includes:
1) Awaiting on top level for async map
2) Fixing a bug in job details when logs are shown undefined.

ttacon commented:
Awesome, thanks!

ttacon commented:
Released as `v2.7.0`

ganeshcse2991 commented:
Thanks :)

Subv commented on src/server/views/dashboard/queueJobsByState.js:
```diff
@@ -96,6 +96,11 @@ async function _html(req, res) {
     jobs = jobs.filter((job) => job);
   } else {
     jobs = await queue[`get${_.capitalize(state)}`](startId, endId);
+    await jobs.map(async (job) => {
```
jobs.map returns an Array, shouldn't you use `await Promise.all(jobs.map(...))` here instead of `await jobs.map(...)`? As it is, you're doing the equivalent of `await Promise.resolve(array)` and thus not actually waiting for the mapped promises to finish

Subv replied:
Could this be related to #204?
